[ios] Update Podfile generation and Pods for CocoaPods 1.6

ide opened the pull request:
# Why

CocoaPods 1.6 changed its Ruby API so the way we access each Pod's `native_target` data structure has changed a little; the template files are now updated to use this new way. This PR still needs to update the xdl dependency in tools-public, which is necessary to get the latest Podfile generator code in XDL.

Closes #4220 (for SDK 33)

# How

Additionally, the way header files are installed has changed and we need to use `@import` syntax for FBAudienceNetwork and GoogleSignIn. To get these to work, the compiler needs to find the frameworks in the framework search paths. We were clobbering the framework search paths with just GoogleMaps paths; this commit just lets us fall back on the default (`$(inherited)`), which includes all the CocoaPods' frameworks.

I also adjusted the indentation a little so the style of the generated Podfile has better indentation.

This PR also updates the version of CocoaPods installed by Nix to 1.6.2 and updates the iOS code to use `@import` where necessary.

# Test Plan

Tested by generating a Podfile, installing the pods successfully, updating the iOS code (coming in another commit) and verifying the app compiles and runs.

ide committed:
[ios] Update Podfile templates and generation code for CocoaPods 1.6

CocoaPods 1.6 changed its Ruby API so the way we access each Pod's `native_target` data structure has changed a little; the template files are now updated to use this new way. This commit uses the updated xdl dependency in tools-public (already updated on master), which is necessary to get the latest Podfile generator code in XDL.

Additionally, the way header files are installed has changed and we need to use `@import` syntax for FBAudienceNetwork and GoogleSignIn. To get these to work, the compiler needs to find the frameworks in the framework search paths. We were clobbering the framework search paths with just GoogleMaps paths; this commit just lets us fall back on the default (`$(inherited)`), which includes all the CocoaPods' frameworks.

I also adjusted the indentation a little so the style of the generated Podfile has better indentation.

This commit also updates the version of CocoaPods installed by Nix to 1.6.2.

As a follow-up commit, we need to update the iOS code to use `@import` where necessary.

Tested by generating a Podfile, installing the pods successfully, updating the iOS code (coming in another commit) and verifying the app compiles and runs.

ide committed:
[nix] Update CocoaPods to 1.6.2

This is in preparation for getting our generated Podfiles compatible with CocoaPods 1.6, which changed the Ruby API and how header files are set up. Note: our Podfile currently relies on CocoaPods 1.5 so this will break things for a short while.

I updated CocoaPods by updating its version in the Gemfile, running `bundler install`, running `bundix` (installed this separately), and then deleting `vendor`.

Tested by running `pod --version` and saw that Nix updated my CocoaPods version and it printed out 1.6.2.

ide committed:
[ios] Use `@import` for FBAudienceNetwork and GoogleSignIn

CocoaPods 1.6 doesn't set up the headers for FBAudienceNetwork and GoogleSignIn and we need to use them as modules instead with the `@import` syntax. This commit changes our code to use `@import`. In a follow-up commit I'll add the updated Pods and am keeping the commits separate so the hand-modified code is separate from the autoinstalled Pods code.

Tested by compiling and running the app with this code and the Pods (to come).

ide committed:
[ios] Update Pods using CocoaPods 1.6 and new Podfile

Regenerated the Pods using the new Podfile that is designed for CocoaPods 1.6 (Ruby API changes and header changes). Compiled and ran the iOS app.

ide commented:
@tsapeta - Could you review this please? It will be easier to review the individual commits (instead of the entire PR), since this PR contains the new Pods from `pod install` as well, which don't need to be review.

I also need to update the version of xdl we use in tools-public, but I can't because npm won't let me publish xdl for permissions reasons (npm looks broken since `npm owner ls` shows me...).

tsapeta commented on tools/ios-tasks.js:
```diff
@@ -489,19 +489,17 @@ async function generatePodfileDepsAsync(
         config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << '${versionName}RCT_DEV=1'
-        # needed for GoogleMaps 2.x
+        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << '${versionName}RCT_ENABLE_INSPECTOR=0'
+        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << '${versionName}ENABLE_PACKAGER_CONNECTION=0'
```
What `ENABLE_PACKAGER_CONNECTION` does? Are you sure it will be versioned? It doesn't start with `RCT` 😉

ide replied:
`ENABLE_PACKAGER_CONNECTION` causes RN to open a web socket connection that causes lots of noisy log messages (https://github.com/expo/expo-cli/commit/24bb708bc51b555ef415a30f2baad16c9e8648e2). I seem to remember that it was for special editor integration that might not be useful in open source.

You're right that it is not versioned -- perhaps we should add it to the versioning script and update the ABI31+32 code in `ios/versioned-react-native`.

ide replied:
I modified the versioning script so that it namespaces `ENABLE_PACKAGER_CONNECTION` (https://github.com/expo/expo/pull/4258) and also edited the old ABI 31 + 32 code to use the namespaces macros.

tsapeta reviewed (approved):
Looks good to me! Thanks for doing it and also for adding `:inhibit_warnings => true` 🙂 - I was going to do that as well 😄 
Just one comment below 👇

ide committed:
[ios] Add 33.0.0 to the ExpoKit template (for CI)

To fix the `shell_app_sim_base_ios` workflow, I added 33.0.0 to the ExpoKit template versions so that the generated ExpoKit Podfile contains the code for CocoaPods 1.6, which we are requiring with SDK 33.

Ran: `gulp ios-shell-app --action build --type simulator --verbose true --skipRepoUpdate --shellAppSdkVersion UNVERSIONED` and got past the Podfile step that usually fails. For some reason my computer ran out of disk (perhaps swap?) when running it but IIRC this is an existing separate issue.
